cheatsheets/Browser_Extension_Vulnerabilities_Cheat_Sheet.md

```diff
@@ -1,43 +1,205 @@
+Follow the Principle of Least Privilege (PoLP) and request only the permissions that are absolutely necessary.
+Use optional permissions whenever possible instead of granting full access upfront.
+Regularly audit and remove any permissions that are no longer needed.
 
-## 4. Insecure Communication
+2. Data Leakage
 
-Data sent over insecure HTTP can be intercepted by attackers on the same network, allowing them to capture sensitive information, such as tokens or personal data.
+## Vulnerability
 
-## 5. Code Injection
+Some extensions unintentionally expose user data by sending browsing activity or personal details to external servers without proper security measures.
 
-If an attacker controls the script URL, they can inject malicious code into the page, leading to data theft or manipulation of the page’s functionality.
+## Example
 
-## 6. Malicious Updates
+chrome.tabs.onUpdated.addListener((tabId, changeInfo, tab) => {
+  if (changeInfo.status === 'complete') {
+    fetch('http://example.com/track', {
+      method: 'POST',
+      body: JSON.stringify({ URL: tab.URL })
+    });
+  }
+});
 
-If the update mechanism is compromised, attackers can push malicious code to users without their knowledge, potentially gaining control over their browsers.
+## Mitigation
 
-## 7. Third-Party Dependencies
+Always use HTTPS for all communications to prevent data interception.
+Limit data collection and be transparent by clearly stating what data is collected in a Privacy Policy.
+Implement user consent mechanisms before collecting or sending any personal data.
 
-An extension relying on outdated third-party libraries may become vulnerable if those libraries have known security flaws that attackers can exploit.
+3. Cross-Site Scripting (XSS)
 
-## 8. Lack of Content Security Policy (CSP)
+## Vulnerability
 
-Without a strong CSP, attackers can inject untrusted content, increasing the risk of XSS and other attacks that manipulate the extension’s behavior.
+If user input is not properly sanitized, attackers can inject malicious scripts into web pages, potentially stealing user data or performing unauthorized actions.
 
-## 9. Insecure Storage
+## Example
 
-If an attacker gains access to the local storage, they can easily retrieve sensitive information, such as tokens or user credentials, leading to unauthorized access.
+let userInput = document.getElementById('input').value;
+document.getElementById('output').innerHTML = userInput; // No sanitization
 
-## 10. Insufficient Privacy Controls
+## Mitigation
 
-Users may be unaware of how their data is being collected or used, leading to potential abuse of their information without consent or awareness.
+Implement Content Security Policy (CSP) to block inline scripts.
+Use libraries like DOMPurify to sanitize user input before displaying it.
+Avoid using innerHTML and instead use textContent to prevent execution of injected scripts.
+
+4. Insecure Communication
```

Comment:
These top level headings should have "##", and the subheadings ("Example", etc) should have "###"